packages/perseus/src/widgets/number-line/__docs__/number-line-interactions-regression.stories.tsx

```diff
@@ -0,0 +1,122 @@
+import {generateTestPerseusItem} from "@khanacademy/perseus-core";
+
+import {ServerItemRendererWithDebugUI} from "../../../../../../testing/server-item-renderer-with-debug-ui";
+
+import type {PerseusRenderer} from "@khanacademy/perseus-core";
+import type {Meta} from "@storybook/react-vite";
+
+// TODO: Add test coverage for hover states on movable points (both closed and open circles).
+
+const meta: Meta = {
+    title: "Widgets/Number Line/Visual Regression Tests/Interactions",
+    component: ServerItemRendererWithDebugUI,
+    tags: ["!autodocs"],
+    parameters: {
+        docs: {
+            description: {
+                component:
+                    "Regression tests for number-line widget interactions. Tests inequality control buttons' behavior which is controlled by Perseus.",
+            },
+        },
+        chromatic: {disableSnapshot: false},
+    },
+};
+
+export default meta;
+
+// Helper function to create number-line question data
+function createNumberLineQuestion(config: {
+    content: string;
+    correctX: number;
+    range: [number, number];
+    initialX?: number;
+    isInequality?: boolean;
+}): PerseusRenderer {
+    return {
+        content: config.content,
+        images: {},
+        widgets: {
+            "number-line 1": {
+                type: "number-line",
+                alignment: "default",
+                static: false,
+                graded: true,
+                options: {
+                    static: false,
+                    labelRange: [null, null],
+                    initialX: config.initialX ?? null,
+                    tickStep: 1,
+                    labelStyle: "decimal",
+                    labelTicks: true,
+                    isInequality: config.isInequality ?? false,
+                    snapDivisions: 2,
+                    range: config.range,
+                    correctRel: "ge",
+                    numDivisions: null,
+                    divisionRange: [1, 10],
+                    correctX: config.correctX,
+                    isTickCtrl: false,
+                },
+                version: {
+                    major: 0,
+                    minor: 0,
+                },
+            },
+        },
+    };
+}
+
+/**
+ * Tests clicking the "Switch direction" button on an inequality.
+ * Starts with x ≥ (ray pointing right), then switches to x ≤ (ray pointing left).
+ * This tests the Perseus-controlled button behavior and ray direction rendering.
+ */
+export const InequalitySwitchDirection = {
```

Comment:
Is it possible to type these with a `Story` type?

Reply:
Okay, did the typing. The play function doesn't work well with the Story type it seems, so I just added an additional comment to ignore typing for the canvas line. I was trying to find another way, but it doesn't look like `screen` methods work. Hopefully it's okay with the extra typing and the expect error line